dd-library-php-setup.php

```diff
@@ -189,22 +227,147 @@ function install($options)
             echo "Installation to '$binaryForLog' was successful\n";
         }
     }
+}
 
-    echo "--------------------------------------------------\n";
-    echo "SUCCESS\n\n";
-    if ($interactive) {
-        echo "Run this script in a non interactive mode adding the following 'php-bin' options:\n";
-        $args = array_merge(
-            $_SERVER["argv"],
-            array_map(
-                function ($el) {
-                    return '--php-bin=' . $el;
-                },
-                array_keys($selectedBinaries)
-            )
+function install_appsec($options, $selectedBinaries)
+{
+    if ($options[OPT_NO_APPSEC]) {
+        return;
+    }
+    if (
+        !isset($options[OPT_APPSEC_FILE]) &&
+        !isset($options[OPT_APPSEC_URL]) &&
+        !isset($options[OPT_APPSEC_VERSION]) ||
+        isset($options[OPT_APPSEC_VERSION]) && $options[OPT_APPSEC_VERSION] == 'latest'
+    ) {
+        $options[OPT_APPSEC_VERSION] = latest_release("dd-appsec-php");
+    }
+
+    echo "Installing appsec\n";
+
+    // Preparing clean tmp folder to extract files
+    $tmpDir = sys_get_temp_dir() . '/dd-appsec';
+    $tarball = "$tmpDir/dd-appsec-php.tar.gz";
+    execute_or_exit(
+        "Cannot create directory '$tmpDir'",
+        "mkdir -p " . escapeshellarg($tmpDir)
+    );
+    execute_or_exit(
+        "Cannot clean '$tmpDir'",
+        "rm -f " . escapeshellarg($tarball)
+    );
+    delete_on_exit($tmpDir);
+
+    // Retrieve the archive to the temporary location
+    if (isset($options[OPT_APPSEC_FILE])) {
+        $tarball = $options[OPT_APPSEC_FILE];
+    } else {
+        $url = isset($options[OPT_APPSEC_URL])
+            ? $options[OPT_APPSEC_URL]
+            : sprintf(
+                'https://github.com/DataDog/dd-appsec-php/releases/' .
+                "download/v%s/dd-appsec-php-%s-amd64.tar.gz",
+                rawurlencode($options[OPT_APPSEC_VERSION]),
+                rawurlencode($options[OPT_APPSEC_VERSION])
+            );
+        download($url, $tarball);
+    }
+
+    $installDir = "{$options[OPT_INSTALL_DIR]}/appsec-" . extract_version_appsec($options, $tarball);
+
+    // copying sources to the final destination
+    execute_or_exit(
+        "Cannot create directory '$installDir'",
+        "mkdir -p " . escapeshellarg($installDir)
+    );
+    execute_or_exit(
+        "Cannot extract files from '$tarball' to directory '$installDir'",
+        'tar --strip-components=1 -C ' . escapeshellarg($installDir) .
+        ' -xf ' . escapeshellarg($tarball)
+    );
+    echo "Installed dd-appsec files to '$installDir'\n";
+
+    // Actual installation
+    foreach ($selectedBinaries as $command => $fullPath) {
+        $binaryForLog = ($command === $fullPath) ? $fullPath : "$command ($fullPath)";
+        echo "Installing appsec to binary: $binaryForLog\n";
+        $phpProperties = ini_values($fullPath);
+
+        if (is_truthy($phpProperties[IS_DEBUG])) {
```

Comment:
Maybe hard fail here in case the user explicilty gave some `--appsec-` flag as arg.